Cabal/Distribution/Utils/Json.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE OverloadedStrings #-}
+-- | Extremely simple JSON helper. Don't do anything too fancy with this!
+
+module Distribution.Utils.Json
+    ( Json(..)
+    , (.=)
+    , renderJson
+    ) where
+
+import Distribution.Compat.Prelude
+import Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy as L
+import qualified Data.ByteString.Lazy.Char8 as C
+import Data.ByteString.Builder
+
+data Json = JsonArray [Json]
+          | JsonBool !Bool
+          | JsonNull
+          | JsonNumber !Int
+          | JsonObject [(ByteString, Json)]
+          | JsonString !ByteString
```

Comment:
What kind of strings are used here. I'd use either `ShortText` or `String` inside `Json` type.

E.g. package-ids, unit-ids are backed up by `ShortText` (and they are in UTF8, so if they don't need escaping they can be used as is).

Yet, currently filepaths are `FilePath` i.e. `String`.

Here are two issues:
- The keys and strings are textual
- double conversions. Thus I'd use `ShortText` for keys in `JsonObject` (as these are most likely static), and would use `String` for `JsonString` (as it looks like most producers are `prettyShow` or alike).
  - I'd like it to be `ShortText`, but if most producers atm are producing `String`, that would be silly. Yet, there could be a comment to eventually change it to `ShortText` (Path abstraction is related: #6667)

Reply:
Is it really a double conversion if this all gets converted to a `ByteString` at the end of the day? As far as I can see for `JsonString`, all the filepaths etc. will go from `String` -> `ByteString`, and then once its in a `ByteString` the builder means it will be cheap to insert it and move it around